swift/extractor/trap/TrapArena.h

```diff
@@ -0,0 +1,70 @@
+//
+// Created by redsun82 on 18.01.22.
+//
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include "swift/extractor/trap/TrapLabel.h"
+
+namespace codeql {
+
+// TrapArena has the responsibilities to
+// * allocate distinct trap #-labels outputting their assignment to '*' or a @-keys to a trap output
+// * forwarding trap entries to said output
+// TODO: split or move these responsibilities into separate classes
+class TrapArena {
+ public:
+  explicit TrapArena(std::ostream& out) : out_{out} {}
+
+  // get a new label for *
+  template <typename Tag>
+  TrapLabel<Tag> getLabel() {
+    auto ret = allocateLabel<Tag>();
+    print(ret, "=*");
```

Comment:
If you write "get", I expect it to do just that, but it's also printing (same for the other `getLabel`s)?

Reply:
good point